Tail.scala

```diff
@@ -0,0 +1,226 @@
+// Author: James Dunn, UC Berkeley (dunn@eecs.berkeley.edu)
+
```

Comment:
All these notes, if relevant should be put into some documentation (preferably not in the code itself) or deleted.